src/main/java/org/socialhistoryservices/delivery/record/service/RecordServiceImpl.java

```diff
@@ -175,35 +194,48 @@ public boolean updateExternalInfo(Record record, boolean hardRefresh) {
             calendar.add(Calendar.DAY_OF_YEAR, -days);
 
             Date lastUpdated = record.getExternalInfoUpdated();
-            if (!hardRefresh && (lastUpdated != null) && lastUpdated.after(calendar.getTime()))
+            if (!hardRefresh && (lastUpdated != null) && lastUpdated.after(calendar.getTime())) {
                 return (record.getParent() != null) && updateExternalInfo(record.getParent(), hardRefresh);
+            }
 
             // We need to update the external info
             String pid = record.getPid();
+
             ExternalRecordInfo eri = lookup.getRecordMetaDataByPid(pid);
             List<ArchiveHoldingInfo> ahi = lookup.getArchiveHoldingInfoByPid(pid);
             Map<String, ExternalHoldingInfo> ehMap = lookup.getHoldingMetadataByPid(pid);
 
             // Update external record info
-            if (record.getExternalInfo() != null)
+            if (record.getExternalInfo() != null) {
                 record.getExternalInfo().mergeWith(eri);
-            else
+            } else {
                 record.setExternalInfo(eri);
+            }
 
             // Update archive holding info
             record.setArchiveHoldingInfo(ahi);
 
+            // add missing units from same container to database
+            if (eri.getContainer() != null) {
+                String listOfUnitIds = lookup.getUnitIdsFromContainer(pid, eri.getContainer(), false);
+                List<Holding> holdings = uriPathToHoldings(listOfUnitIds, false);
```

Comment:
Not a bad idea to handle the other records in the same container here. But don't use `uriPathToHoldings` for this purpose. As it says, this is specifically for URIs on the controller level.